### ISSUE TITLE
Why not share your work with Ngx-tiptap?

### ISSUE THREAD

**Issue by GitHubish** [HuiiBuh/ngx-tiptap-editor#9]
I just discovered your repository while working on Tiptap for a while. There is a very good project called [Ngx-tiptap](https://github.com/sibiraj-s/ngx-tiptap) maintained by @sibiraj-s who is very active and does a great job. Maybe you could work together?

**Comment by HuiiBuh** [comment]
A few reasons:
- Firstly, I think that the approach of him is more focused on giving you the right set of tooling to create an editor for yourself and not to give people the opportunity to create a full editor
- We have very different design philosophies
- Sibiraj does not escape the NG zone as aggressive as I do, so the performance will be worse, because of unnecessary change detection
- My Implementation is far more complex, because I change control states without the angular CD system, which he does not have to, because his scope (as said) is only to provide tools

So these are my reasons. Are you missing any features from my implementation, or do you fear that I will abandon the project?

**Comment by GitHubish** [comment]
I understand your point of view. I had noticed some differences that you stated, where I have some concerns with the current Ngx-Tiptap. The reason why I asked this question beyond the interest in my opinion to pool these forces to create a community around Tiptap is indeed to choose the right technology so that the implementation can be followed in time and that people can contribute (including me as soon as I feel technically able to do so). 

I have not yet been able to test your implementation (besides the demo is not yet deployed ?) and it seems to me necessarily a little complex to switch from Ngx-tiptap to your implementation to have the equivalent. Do you have any idea how to display a readonly version (let's say a viewer) allowing to display the rendering according to the JSON model of prosemirror (with extensions) in a performant way by reusing Angular extensions as you propose as well as Ngx-tiptap. I don't see how to do this with your solution either. Currently I use the `generateHTML()`  method of `tiptap/html` but of course it does not allow to reuse the extensions previously created for the editor part. 

Thank you for your answers, I will try to look at it more closely.

**Comment by HuiiBuh** [comment]
Hi, I understand your incentive to create a community around tiptap. I too
think that this would profit the project.
Well if you have a look at the docs folder. you can see that one of the
parameters of the editor is the parameter _editable_. With this parameter
you can disable the editing functionality.
If you want to only display the content you can do this with the
tip-editor-preview component, but you have to pass the extensions.
There is simply no other way to archive this.

**Comment by GitHubish** [comment]
In my case I want to display several hundreds of contents on the same view. Using the editor in read-only mode is not possible today with Ngx-tiptap in terms of performance - the rendering time is too important. Is it possible with your read-only editor? 

Concerning tip-editor-preview I think this solution does not work with AngularExtensions that we could have created from what I see. Am I wrong?

**Comment by HuiiBuh** [comment]
No, the read only view will always have to much overhead for such a thing.
Currently you are right, but this is only a small change and already on my radar (see #8).
This should not take any time to implement, as you only have to pass the `ExtensionBuilder` array which you pass to the editor and pass it to the preview as well...

**Comment by HuiiBuh** [comment]
See 
https://github.com/HuiiBuh/ngx-tiptap-editor/blob/main/projects/ngx-tiptap-editor/src/lib/components/editor/editor.component.ts#L73
and 
https://github.com/HuiiBuh/ngx-tiptap-editor/blob/main/projects/ngx-tiptap-editor/src/lib/components/editor/editor.component.ts#L177

**Comment by GitHubish** [comment]
Ok thanks @HuiiBuh I understand, I will be interested in this feature anyway. I will evaluate the complexity to eventually switch from ngx-tiptap to your editor.
